src/modules/sync/advancedDiff.ts

```diff
@@ -0,0 +1,720 @@
+): string => {
+  const template = readFileSync(resolve(templatePath), "utf-8");
+  const resolvePlaceHolders = (template: string): string => {
+    const processedTemplate = template.replace(
+      /{{.*?}}/g,
+      match => rendererMap.get(match)?.(diffData) ?? match,
+    );
+
+    return processedTemplate === template
+      ? template
+      : resolvePlaceHolders(processedTemplate);
+  };
+
+  return resolvePlaceHolders(template);
+};
+
+const resolveOutputPath = (outputPath: string) => {
+  const hasExtension = !!resolve(outputPath).split("/").pop()?.includes(".");
+
+  return hasExtension
+    ? resolve(outputPath)
+    : resolve(
+      outputPath,
+      `diff_${new Date().toISOString().replace(/[:.-]/g, "_")}.html`,
+    );
+};
+
+const createOutputFile = (path: string, content: string, logOptions: LogOptions) => {
+  try {
+    logInfo(
+      logOptions,
+      "standard",
+      chalk.yellow(`Generating a diff file at ${path}`),
+    );
+    writeFileSync(path, content);
+  } catch (err) {
+    logError(logOptions, `Failed writing a diff file: ${JSON.stringify(err)}`);
+    process.exit(1);
+  }
+};
+
+const openOutputFile = (path: string, logOptions: LogOptions) => {
+  logInfo(
+    logOptions,
+    "standard",
+    chalk.green(`Diff file created successfully. Opening...`),
+  );
+  open(path).catch((err) =>
+    logError(
+      logOptions,
+      `Failed to open the file: ${path}\nMessage: ${JSON.stringify(err)}`,
+    )
+  );
+};
+
+const createOutputDirectory = (path: string, logOptions: LogOptions) => {
+  try {
+    logInfo(
+      logOptions,
+      "standard",
+      chalk.yellow(`Creating a directory '${path}'`),
+    );
+    mkdirSync(path, { recursive: true });
+  } catch (err) {
+    logError(logOptions, `Failed to create directory '${path}': ${JSON.stringify(err)}`);
+    process.exit(1);
+  }
+};
+
+const renderMoveOpPosition = (patchOp: MovePatchOperation) =>
+  "after" in patchOp
+    ? `after <strong>${patchOp.after.codename}</strong>`
+    : "before" in patchOp
+    ? `before <strong>${patchOp.before.codename}</strong>`
+    : `under <strong>${patchOp.under.codename}</strong>`;
+
+const renderAddIntoOpPosition = (patchOp: AddIntoPatchOperation) =>
+  patchOp.after
+    ? `after <strong>${[patchOp.after.codename]}</strong>`
+    : patchOp.before
+    ? `before <strong>${[patchOp.before.codename]}</strong>`
+    : "";
+
+const renderRemoveOpDetail = (patchOp: RemovePatchOperation) =>
+  `${modifierMap.get("remove")}${
+    getEntityPathRenderer(
+      removeEntityPathRenderers,
+      patchOp.path,
+    )
+  } removed`;
+
+const renderMoveOpDetail = (patchOp: MovePatchOperation) =>
+  `${modifierMap.get("move")}${
+    getEntityPathRenderer(
+      moveEntityPathRenderers,
+      patchOp.path,
+    )
+  } moved ${renderMoveOpPosition(patchOp)}`;
+
+const renderAddIntoOpDetail = (patchOp: AddIntoPatchOperation) =>
+  `${modifierMap.get("addInto")}${
+    getEntityPathRenderer(
+      addEntityPathRenderers,
```

Comment:
this can be one liner. Sometimes you have to navigate the `dprint` to better looking solution by  adjusting manually :D if your vscode still don't use dprint adjus in vscode settings.

something like CTRL + SHIFT + P and type settings

Reply:
well, dprint met us halfway through and turned them into 3-liners after some friendly guidance